.github/workflows/ci.yml

```
@@ -41,6 +41,8 @@ jobs:
 
       - run: cargo test --locked --all-features --all-targets
       - run: cargo test --locked --all-features --doc
+      - run: cd bon && cargo test --locked --no-default-features --features=
+      - run: cd bon && cargo test --locked --no-default-features --features=alloc
```

Let's include doc tests in these test runs
```suggestion
      - run: cd bon && cargo test --locked --all-targets --doc --no-default-features --features=
      - run: cd bon && cargo test --locked --all-targets --doc --no-default-features --features=alloc
```

---

I think that doc tests are run by default:
```
❭ cd bon && cargo test --locked --no-default-features --features=
    Finished `test` profile [unoptimized + debuginfo] target(s) in 0.06s
     Running unittests src/lib.rs (/home/danielschemmel/projects/bon/target/debug/deps/bon-0551bf49c71843ca)

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s

     Running tests/integration/main.rs (/home/danielschemmel/projects/bon/target/debug/deps/integration-d2abf59b71eb5bec)

running 22 tests
test builder_on_fn::const_function ... ok
test builder_on_fn::expose_positional_fn::method_new_doesnt_require_a_value_for_name ... ok
test builder_on_fn::constructor ... ok
test builder_on_fn::expose_positional_fn::simple ... ok
test builder_on_fn::expose_positional_fn::with_nested_params ... ok
test builder_on_fn::generics_with_lifetimes ... ok
test builder_on_fn::impl_block_ty_contains_a_reference ... ok
test builder_on_fn::impl_block_with_self_in_const_generics ... ok
test builder_on_fn::leading_underscore_is_stripped ... ok
test builder_on_fn::lifetime_elision ... ok
test builder_on_fn::mut_fn_params ... ok
test builder_on_fn::raw_identifiers ... ok
test builder_on_fn::receiver ... ok
test builder_on_fn::receiver_is_non_default ... ok
test builder_on_fn::receiver_with_lifetimes ... ok
test builder_on_fn::self_in_a_bunch_of_places ... ok
test builder_on_fn::self_only_generic_param ... ok
test builder_on_fn::types_not_implementing_default ... ok
test builder_on_fn::unsafe_func ... ok
test builder_on_struct::raw_identifiers ... ok
test builder_on_struct::smoke ... ok
     Locking 1 package to latest compatible version
      Adding bon-tests v0.0.0 (/home/danielschemmel/projects/bon/target/tests/trybuild/bon)
    Checking bon-tests v0.0.0 (/home/danielschemmel/projects/bon/target/tests/trybuild/bon)
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.07s


test tests/integration/ui/compile_fail/misc.rs ... ok


test ui::ui ... ok

test result: ok. 22 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.18s

   Doc-tests bon

running 2 tests
test bon/src/lib.rs - arr (line 58) ... ok
test bon/src/lib.rs - (line 23) ... ok

test result: ok. 2 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.09s
```

The last two tests are the doc tests (this run is nostd without alloc, so there are only two doc tests).

Either way, your suggested command does not have the intended result:
```
❭ cd bon && cargo test --locked --all-targets --doc --no-default-features --features=
error: Can't mix --doc with other target selecting options
```

---

Aha, makes sense